lib/auth/integration/integrationv1/awsoidc.go

```diff
@@ -249,6 +249,65 @@ func (s *AWSOIDCService) ListEICE(ctx context.Context, req *integrationpb.ListEI
 	}, nil
 }
 
+// CreateEICE creates multiple EC2 Instance Connect Endpoint using the provided Subnets and Security Group IDs.
+func (s *AWSOIDCService) CreateEICE(ctx context.Context, req *integrationpb.CreateEICERequest) (*integrationpb.CreateEICEResponse, error) {
+	authCtx, err := s.authorizer.Authorize(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if err := authCtx.CheckAccessToKind(types.KindIntegration, types.VerbUse); err != nil {
+		return nil, trace.Wrap(err)
+	}
```

Comment:
Is this granular enough? I think as is, if the client can `use` any integration, they are allowed to create EICE endpoints?

I assume this will be called by the proxy which basically needs to be able to use any integration for now, but as we add new integrations I think it would be nice to give admins the power to restrict users to specific integrations something like
```yaml
allow:
  rules:
    resources: [integration]
    verbs: [use]
    where: 'name == "example-integration"'
```

I'll approve this PR since it's kind of tangential and there's already other endpoints that do the same, but I think it would be best to add server-side checking for this as early as we can so that supported auth versions are likely to check the `where` expressions whenever we eventually want to use them

Reply:
Totally agreed 👍 
This was discussed in the RFD but we ended up with the most simple approach.
Even the `use` verb was not consensual.
https://github.com/gravitational/teleport/pull/24256#discussion_r1163043822

Honestly, I think we need to go even further.
We should support integration names and even integration actions.
To expand on your example:
```yaml
allow:
  rules:
    resources: [integration]
    verbs: [use]
    where: 'name == "example-integration" && action == "awsoidc-create-eice"'
```

I also agree that this PR should not be the one introducing that.
I'll keep this topic under my radar and possibly come up with a proposal.